### ISSUE TITLE
Why loss become zero?

### ISSUE THREAD

**Issue by Labmem009** [wwxu21/CUT#2]
During my first round online alignment, the loss:
{'loss': 1.6736, 'learning_rate': 4.000000000000001e-06, 'epoch': 0.02}                                         
{'loss': 0.0, 'learning_rate': 8.000000000000001e-06, 'epoch': 0.03}                                            
{'loss': 0.0, 'learning_rate': 1.2e-05, 'epoch': 0.05}                                                          
{'loss': 0.0, 'learning_rate': 1.6000000000000003e-05, 'epoch': 0.07}                                           
{'loss': 0.0, 'learning_rate': 2e-05, 'epoch': 0.08}                                                            
{'loss': 0.0, 'learning_rate': 2.4e-05, 'epoch': 0.1}                                                           
{'loss': 0.0, 'learning_rate': 2.8000000000000003e-05, 'epoch': 0.11}                                           
{'loss': 0.0, 'learning_rate': 3.2000000000000005e-05, 'epoch': 0.13}                                           
{'loss': 0.0, 'learning_rate': 3.6e-05, 'epoch': 0.15}                                                          
{'loss': 0.0, 'learning_rate': 4e-05, 'epoch': 0.16}                                                            
{'loss': 0.0, 'learning_rate': 4.4000000000000006e-05, 'epoch': 0.18}                                           
{'loss': 0.0, 'learning_rate': 4.8e-05, 'epoch': 0.2}                                                           
{'loss': 0.0, 'learning_rate': 5.2000000000000004e-05, 'epoch': 0.21}                                           
{'loss': 0.0, 'learning_rate': 5.6000000000000006e-05, 'epoch': 0.23}                                           
{'loss': 0.0, 'learning_rate': 6e-05, 'epoch': 0.24}                                                            
{'loss': 0.0, 'learning_rate': 6.400000000000001e-05, 'epoch': 0.26}                                            
{'loss': 0.0, 'learning_rate': 6.800000000000001e-05, 'epoch': 0.28}                                            
{'loss': 0.0, 'learning_rate': 7.2e-05, 'epoch': 0.29}                                                          
{'loss': 0.0, 'learning_rate': 7.6e-05, 'epoch': 0.31}                                                          
{'loss': 0.0, 'learning_rate': 8e-05, 'epoch': 0.33}                                                            
{'loss': 0.0, 'learning_rate': 8.4e-05, 'epoch': 0.34}                                                          
{'loss': 0.0, 'learning_rate': 8.800000000000001e-05, 'epoch': 0.36}                                            
{'loss': 0.0, 'learning_rate': 9.200000000000001e-05, 'epoch': 0.37}                                            
{'loss': 0.0, 'learning_rate': 9.6e-05, 'epoch': 0.39}                                                          
{'loss': 0.0, 'learning_rate': 0.0001, 'epoch': 0.41}                                                           
{'loss': 0.0, 'learning_rate': 0.00010400000000000001, 'epoch': 0.42}                                           
{'loss': 0.0, 'learning_rate': 0.00010800000000000001, 'epoch': 0.44}                                           
{'loss': 0.0, 'learning_rate': 0.00011200000000000001, 'epoch': 0.46}                                           
{'loss': 0.0, 'learning_rate': 0.000116, 'epoch': 0.47}                                                         
{'loss': 0.0, 'learning_rate': 0.00012, 'epoch': 0.49}                                                          
{'loss': 0.0, 'learning_rate': 0.000124, 'epoch': 0.5}                                                          
{'loss': 0.0, 'learning_rate': 0.00012800000000000002, 'epoch': 0.52}                                           
{'loss': 0.0, 'learning_rate': 0.000132, 'epoch': 0.54}                                                         
{'loss': 0.0, 'learning_rate': 0.00013600000000000003, 'epoch': 0.55}                                           
{'loss': 0.0, 'learning_rate': 0.00014, 'epoch': 0.57}                                                          
{'loss': 0.0, 'learning_rate': 0.000144, 'epoch': 0.59}                                                         
{'loss': 0.0, 'learning_rate': 0.000148, 'epoch': 0.6}                                                          
{'loss': 0.0, 'learning_rate': 0.000152, 'epoch': 0.62}                                                         
{'loss': 0.0, 'learning_rate': 0.00015600000000000002, 'epoch': 0.63}                                           
{'loss': 0.0, 'learning_rate': 0.00016, 'epoch': 0.65}                                                          
{'loss': 0.0, 'learning_rate': 0.000164, 'epoch': 0.67}                                                         
{'loss': 0.0, 'learning_rate': 0.000168, 'epoch': 0.68}                                                         
{'loss': 0.0, 'learning_rate': 0.000172, 'epoch': 0.7}                                                          
{'loss': 0.0, 'learning_rate': 0.00017600000000000002, 'epoch': 0.72}                                           
{'loss': 0.0, 'learning_rate': 0.00018, 'epoch': 0.73}                                                          
{'loss': 0.0, 'learning_rate': 0.00018400000000000003, 'epoch': 0.75}                                           
{'loss': 0.0, 'learning_rate': 0.000188, 'epoch': 0.76}                                                         
{'loss': 0.0, 'learning_rate': 0.000192, 'epoch': 0.78}                                                         
{'loss': 0.0, 'learning_rate': 0.000196, 'epoch': 0.8}                                                          
{'loss': 0.0, 'learning_rate': 0.0002, 'epoch': 0.81}                                                           
{'loss': 0.0, 'learning_rate': 0.00020400000000000003, 'epoch': 0.83}                                           
{'loss': 0.0, 'learning_rate': 0.00020800000000000001, 'epoch': 0.85}                                           
{'loss': 0.0, 'learning_rate': 0.00021200000000000003, 'epoch': 0.86}                                           
{'loss': 0.0, 'learning_rate': 0.00021600000000000002, 'epoch': 0.88}                                           
{'loss': 0.0, 'learning_rate': 0.00022000000000000003, 'epoch': 0.89}                                           
{'loss': 0.0, 'learning_rate': 0.00022400000000000002, 'epoch': 0.91}                                           
{'loss': 0.0, 'learning_rate': 0.00022799999999999999, 'epoch': 0.93}                                           
{'loss': 0.0, 'learning_rate': 0.000232, 'epoch': 0.94}                                                         
{'loss': 0.0, 'learning_rate': 0.000236, 'epoch': 0.96}                                                         
{'loss': 0.0, 'learning_rate': 0.00024, 'epoch': 0.98}                                                          
{'loss': 0.0, 'learning_rate': 0.000244, 'epoch': 0.99}
I use the script in README, why the loss become 0?
Looking forward for help

**Comment by wwxu21** [comment]
Loss 0 means there is a `nan` in your loss. You can reduce the `weight_unlike` to 0.5 or even lower.
In online learning, the response is generated by the model to be aligned. Therefore the loss is usually not very large (usually < 1). It seems that the initial loss of your is extremely large. Please make sure you are conducting online alignment. 
I rerun the README script with llama2-13b-chat for your reference. Here is the training log:
{'loss': 0.3341, 'learning_rate': 4.000000000000001e-06, 'epoch': 0.02}                                       
{'loss': 0.3462, 'learning_rate': 8.000000000000001e-06, 'epoch': 0.04}                                       
{'loss': 0.4011, 'learning_rate': 1.2e-05, 'epoch': 0.07}                                                     
{'loss': 0.3018, 'learning_rate': 1.6000000000000003e-05, 'epoch': 0.09}                                      
{'loss': 0.3577, 'learning_rate': 2e-05, 'epoch': 0.11}                                                       
{'loss': 0.2905, 'learning_rate': 2.4e-05, 'epoch': 0.13}                                                     
{'loss': 0.2539, 'learning_rate': 2.8000000000000003e-05, 'epoch': 0.16}                                      
{'loss': 0.312, 'learning_rate': 3.2000000000000005e-05, 'epoch': 0.18}                                       
{'loss': 0.2936, 'learning_rate': 3.6e-05, 'epoch': 0.2}                                                      
{'loss': 0.319, 'learning_rate': 4e-05, 'epoch': 0.22}                                                        
{'loss': 0.3109, 'learning_rate': 4.4000000000000006e-05, 'epoch': 0.24}                                      
{'loss': 0.3698, 'learning_rate': 4.8e-05, 'epoch': 0.27}                                                     
{'loss': 0.2499, 'learning_rate': 5.2000000000000004e-05, 'epoch': 0.29}                                      
{'loss': 0.3518, 'learning_rate': 5.6000000000000006e-05, 'epoch': 0.31}                                      
{'loss': 0.2139, 'learning_rate': 6e-05, 'epoch': 0.33}                                                       
{'loss': 0.2363, 'learning_rate': 6.400000000000001e-05, 'epoch': 0.36}                                       
{'loss': 0.3818, 'learning_rate': 6.800000000000001e-05, 'epoch': 0.38}                                       
{'loss': 0.1942, 'learning_rate': 7.2e-05, 'epoch': 0.4}                                                      
{'loss': 0.2516, 'learning_rate': 7.6e-05, 'epoch': 0.42}                                                     
{'loss': 0.2094, 'learning_rate': 8e-05, 'epoch': 0.44}                                                       
{'loss': 0.2088, 'learning_rate': 8.4e-05, 'epoch': 0.47}                                                     
{'loss': 0.2223, 'learning_rate': 8.800000000000001e-05, 'epoch': 0.49}                                       
{'loss': 0.2114, 'learning_rate': 9.200000000000001e-05, 'epoch': 0.51}                                       
{'loss': 0.2106, 'learning_rate': 9.6e-05, 'epoch': 0.53}                                                     
{'loss': 0.3528, 'learning_rate': 0.0001, 'epoch': 0.56}                                                      
{'loss': 0.1928, 'learning_rate': 0.00010400000000000001, 'epoch': 0.58}                                      
{'loss': 0.179, 'learning_rate': 0.00010800000000000001, 'epoch': 0.6}                                        
{'loss': 0.2303, 'learning_rate': 0.00011200000000000001, 'epoch': 0.62}                                      
{'loss': 0.214, 'learning_rate': 0.000116, 'epoch': 0.64}                                                     
{'loss': 0.2143, 'learning_rate': 0.00012, 'epoch': 0.67}                                                     
{'loss': 0.2467, 'learning_rate': 0.000124, 'epoch': 0.69}                                                    
{'loss': 0.2316, 'learning_rate': 0.00012800000000000002, 'epoch': 0.71}                                      
{'loss': 0.1977, 'learning_rate': 0.000132, 'epoch': 0.73}                                                    
{'loss': 0.1905, 'learning_rate': 0.00013600000000000003, 'epoch': 0.76}                                      
{'loss': 0.16, 'learning_rate': 0.00014, 'epoch': 0.78}                                                       
{'loss': 0.1563, 'learning_rate': 0.000144, 'epoch': 0.8}                                                     
{'loss': 0.2037, 'learning_rate': 0.000148, 'epoch': 0.82}                                                    
{'loss': 0.2065, 'learning_rate': 0.000152, 'epoch': 0.84}                                                    
{'loss': 0.2428, 'learning_rate': 0.00015600000000000002, 'epoch': 0.87}                                      
{'loss': 0.1608, 'learning_rate': 0.00016, 'epoch': 0.89}                                                     
{'loss': 0.1886, 'learning_rate': 0.000164, 'epoch': 0.91}                                                    
{'loss': 0.2447, 'learning_rate': 0.000168, 'epoch': 0.93}                                                    
{'loss': 0.2278, 'learning_rate': 0.000172, 'epoch': 0.96}                                                    
{'loss': 0.1263, 'learning_rate': 0.00017600000000000002, 'epoch': 0.98}                                      
{'loss': 0.221, 'learning_rate': 0.00018, 'epoch': 1.0}                                                       
{'train_runtime': 156.1767, 'train_samples_per_second': 2.292, 'train_steps_per_second': 0.288, 'train_loss': 0.24934316674868265, 'epoch': 1.0}
